### PR TITLE
feat(js,client): enable user endpoint in js core and client sdks

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Enable user endpoint in JS core and client SDKs

## Testing

- [x] Locally tested
- [x] Passed all UTs

### CODE REVIEW

File: packages/js/src/core/user-info.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  role_names?: string[];
+};
+
+export const fetchUserInfo = async (
+  userInfoEndpoint: string,
+  accessToken: string,
+  requester: Requester
+): Promise<UserInfoResponse> =>
+  requester<UserInfoResponse>(userInfoEndpoint, {
+    headers: { Authorization: `Bearer ${accessToken}` },
```
Comment (wangsijie) [inline packages/js/src/core/user-info.ts:17]: In node sdk, we mutate the "headers", would you mind checking if there is a conflict issue?

Comment (wangsijie) [reply packages/js/src/core/user-info.ts:17]: for appSecret

Comment (charIeszhao) [reply packages/js/src/core/user-info.ts:17]: Hmmm, I think you're right. Should move this to browser SDK

Comment (wangsijie) [reply packages/js/src/core/user-info.ts:17]: But seems OK for now, this endpoint does not require `appSecret` to work?

Comment (charIeszhao) [reply packages/js/src/core/user-info.ts:17]: No, it does not require appSecret. Yeah, I thought about this couple days ago but forgot to update the comment. So I guess it's fine for now

**Review by wangsijie — APPROVED** [review]
LGTM

**Pull request merged**